woocommerce/woocommerce
# Fix Description Length Preventing REST API Key Creation

### All Submissions:






### Changes proposed in this Pull Request:



Closes #30594.

This PR implements the changes to the API Key description value (generated for keys generated via the REST API) suggested in #30594 as well as basic safeguards from causing the same error using the "Add key" UI in settings. It also introduces a user facing error message if the API Key creation fails via the admin interface:

<img width="835" alt="Screen Shot 2021-10-07 at 5 03 11 PM" src="https://user-images.githubusercontent.com/63922/136462417-98805f9e-5fcd-40e8-aa04-527412800072.png">

With this PR, the endpoint generated key description no longer requires translation and will truncate the `app_name` at `170` characters. 

### How to test the changes in this Pull Request:

If your test store uses a self-signed SSL certificate, add this filter:

```php
add_filter( 'http_request_args', function( $args ) {
        $args['sslverify'] = false;
        return $args;
} );
```

1. Go to the following URL for your store: `http://[STORE_DOMAIN]/wc-auth/v1/authorize?app_name=[APP_NAME]&scope=read_write&user_id=1&return_url=[STORE_DOMAIN]&callback_url=[STORE_DOMAIN]` (where [STORE_DOMAIN] is your test store, and [APP_NAME] is at least 200 characters long)
2. Follow the "approve" link
3. Verify that the key is created and the description contains the truncated `app_name`, the text ` - API`, and the current datetime.
4. Go to WooCommerce > Settings > Advanced > REST API
5. Click "Add key"
6. Attempt to enter a description longer than 200 characters (paste in)
7. Click "Generate API key"
8. Verify the key was created and the description truncated at 200 chars
9. Go to WooCommerce > Settings > Advanced > REST API
10. Click "Add key"
11. Use the developer tools to inspect the description field and remove the `maxlength` attribute
12. Paste in a description longer than 200 characters
13. Click "Generate API key"
14. Verify an error message is shown

### Other information:




### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

Fix bug when creating REST API keys with very long descriptions.

### FOR PR REVIEWER ONLY:
